backend/LexBoxApi/Services/UserService.cs

```diff
@@ -16,6 +16,16 @@ await dbContext.Users.Where(u => u.Id == id)
             .ExecuteUpdateAsync(c => c.SetProperty(u => u.LastActive, DateTimeOffset.UtcNow));
     }
 
+    public async Task UpdatePasswordStrength(LexCore.Entities.User user, LexCore.LoginRequest loginRequest)
+    {
+        if (user.PasswordStrength is null && !loginRequest.PreHashedPassword)
+        {
+            var strength = Zxcvbn.Core.EvaluatePassword(loginRequest.Password);
+            user.PasswordStrength = strength.Score;
+            await dbContext.SaveChangesAsync();
```

Comment:
it would be best not to take a user from an unknown source like this and modify it then call SaveChanges, in this case I'd copy `UpdateUserLastActive` where it just explicitly sets the value and executes an update.

Reply:
My thinking was that I wanted to avoid making another unnecessary DB query when I already have the User entity. Since EvaluatePassword costs CPU time, I only want to run it when the user's password strength is NULL in the DB, and since every time I'd want to call this method I already have the user entity, why not use it? But I'll switch it to use a GUID instead, and run another DB query first.

Reply:
Done, and that allowed me to back out the changes I had made to LexAuthService.Login's return type. So although this runs one extra DB query, it's a SELECT EXISTS query so it should be fast for the database to run. And now we've made fewer code changes, which means less chance of bugs and/or merge conflicts around this code.